fe/fe-core/src/main/java/com/starrocks/system/HistoricalNodeMgr.java

```diff
@@ -34,55 +41,70 @@
 
 public class HistoricalNodeMgr {
     private static final Logger LOG = LogManager.getLogger(HistoricalNodeMgr.class);
-    // TODO(ComputeResource): support more better compute resource acquiring.
-    private ConcurrentHashMap<String, HistoricalNodeSet> whToComputeNodeIds;
+    // Use a string as the map key to be compatible with the old image data format.
+    // This key consists of warehouseId and workerGroupId, such as: "warehouseId_workerGroupId".
+    private ConcurrentHashMap<String, HistoricalNodeSet> computeResourceToNodeSet;
 
     public HistoricalNodeMgr() {
-        whToComputeNodeIds = new ConcurrentHashMap<>();
+        computeResourceToNodeSet = new ConcurrentHashMap<>();
     }
 
-    public void updateHistoricalBackendIds(List<Long> backendIds, long currentTime, String warehouse) {
-        HistoricalNodeSet nodeSet = whToComputeNodeIds.computeIfAbsent(warehouse, k -> new HistoricalNodeSet());
+    public void updateHistoricalBackendIds(long warehouseId, long workerGroupId, List<Long> backendIds, long currentTime) {
+        String resourceKey = computeResourceIdToStr(warehouseId, workerGroupId);
+        HistoricalNodeSet nodeSet = computeResourceToNodeSet.computeIfAbsent(resourceKey, k -> new HistoricalNodeSet());
         nodeSet.updateHistoricalBackendIds(backendIds, currentTime);
     }
 
-    public void updateHistoricalComputeNodeIds(List<Long> computeNodeIds, long currentTime, String warehouse) {
-        HistoricalNodeSet nodeSet = whToComputeNodeIds.computeIfAbsent(warehouse, k -> new HistoricalNodeSet());
+    public void updateHistoricalComputeNodeIds(long warehouseId, long workerGroupId, List<Long> computeNodeIds,
+                                               long currentTime) {
+        String resourceKey = computeResourceIdToStr(warehouseId, workerGroupId);
+        HistoricalNodeSet nodeSet = computeResourceToNodeSet.computeIfAbsent(resourceKey, k -> new HistoricalNodeSet());
         nodeSet.updateHistoricalComputeNodeIds(computeNodeIds, currentTime);
     }
 
-    public ImmutableList<Long> getHistoricalBackendIds(String warehouse) {
-        HistoricalNodeSet nodeSet = whToComputeNodeIds.get(warehouse);
+    public ImmutableList<Long> getHistoricalBackendIds(long warehouseId, long workerGroupId) {
```

Comment:
shall leverage ComputeResource concept for the interface. It will be OK to break down into warehouseId + workerGroupId in the detailed implementation.